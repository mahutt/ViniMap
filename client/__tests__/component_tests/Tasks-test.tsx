import React from 'react';
import { render } from '@testing-library/react-native';
import TasksScreen from '@/app/(tabs)/tasks';

jest.mock('@/components/TaskCard', () => 'TaskCard');

describe('TasksScreen Component', () => {
  it('matches snapshot when there are no tasks', () => {
    const { toJSON } = render(<TasksScreen />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('matches snapshot when there are tasks', () => {
    jest.spyOn(React, 'useState').mockReturnValueOnce([
      [
        { id: '1', text: 'Task 1', coordinates: [0, 0] },
        { id: '2', text: 'Task 2', coordinates: [0, 0] },
      ],
      jest.fn(),
    ]);

    const { toJSON } = render(<TasksScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
});
