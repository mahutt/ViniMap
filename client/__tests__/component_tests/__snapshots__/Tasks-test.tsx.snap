// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksScreen Component matches snapshot when there are no tasks 1`] = `
<View
  style={
    {
      "backgroundColor": "#E8EAED",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      {
        "paddingTop": 80,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
            "paddingBottom": 20,
          }
        }
      >
        Today's Tasks
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "color": "#852C3A",
              "fontSize": 16,
              "paddingRight": 10,
            }
          }
        >
           Undo
        </Text>
      </View>
    </View>
    <RCTScrollView
      style={
        {
          "height": 500,
        }
      }
    >
      <View>
        <Text
          style={
            {
              "color": "gray",
              "marginTop": 20,
              "textAlign": "center",
            }
          }
        >
          No tasks yet.
        </Text>
      </View>
    </RCTScrollView>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
            "marginRight": 10,
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Task name"
          style={
            {
              "backgroundColor": "#FFF",
              "borderRadius": 10,
              "marginBottom": 10,
              "padding": 10,
            }
          }
          value=""
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Location"
          style={
            {
              "backgroundColor": "#FFF",
              "borderRadius": 10,
              "marginBottom": 10,
              "padding": 10,
            }
          }
          value=""
        />
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#852C3A",
            "borderRadius": 10,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": 50,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 24,
            }
          }
        >
          +
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`TasksScreen Component matches snapshot when there are tasks 1`] = `
<View
  style={
    {
      "backgroundColor": "#E8EAED",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      {
        "paddingTop": 80,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
            "paddingBottom": 20,
          }
        }
      >
        Today's Tasks
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "color": "#852C3A",
              "fontSize": 16,
              "paddingRight": 10,
            }
          }
        >
           Undo
        </Text>
      </View>
    </View>
    <RCTScrollView
      style={
        {
          "height": 500,
        }
      }
    >
      <View>
        <TaskCard
          onDelete={[Function]}
          text="Task 1"
        />
        <TaskCard
          onDelete={[Function]}
          text="Task 2"
        />
      </View>
    </RCTScrollView>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
            "marginRight": 10,
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Task name"
          style={
            {
              "backgroundColor": "#FFF",
              "borderRadius": 10,
              "marginBottom": 10,
              "padding": 10,
            }
          }
          value=""
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Location"
          style={
            {
              "backgroundColor": "#FFF",
              "borderRadius": 10,
              "marginBottom": 10,
              "padding": 10,
            }
          }
          value=""
        />
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#852C3A",
            "borderRadius": 10,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": 50,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 24,
            }
          }
        >
          +
        </Text>
      </View>
    </View>
  </View>
</View>
`;
